karyon3-eureka/src/main/java/com/netflix/karyon/eureka/KaryonEurekaClientConfig.java

```diff
@@ -0,0 +1,192 @@
+package com.netflix.karyon.eureka;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.google.common.base.Objects;
+import com.google.inject.Inject;
+import com.netflix.archaius.Config;
+import com.netflix.archaius.annotations.Configuration;
+import com.netflix.archaius.annotations.ConfigurationSource;
+import com.netflix.discovery.EurekaClientConfig;
+
+@Configuration(prefix="eureka")
+@ConfigurationSource("eureka-client")
+public class KaryonEurekaClientConfig implements EurekaClientConfig {
+    public static final String DEFAULT_ZONE = "defaultZone";
+    
+    private final Config config;
+    private final String defaultRegion;
+
+    @Inject
+    KaryonEurekaClientConfig(Config config) {
+        this.defaultRegion = config.getString("@region", null);
+        
+        this.config = config.getPrefixedView("eureka");
+    }
+    
+    public int getRegistryFetchIntervalSeconds() {
+        return config.getInteger("client.refresh.interval", 30);
+    }
+
+    public int getInstanceInfoReplicationIntervalSeconds() {
+        return config.getInteger("appinfo.replicate.interval", 30);
+    }
+
+    public int getInitialInstanceInfoReplicationIntervalSeconds() {
+        return config.getInteger("appinfo.initial.replicate.time", 40);
+    }
+
+    public int getEurekaServiceUrlPollIntervalSeconds() {
+        return config.getInteger("serviceUrlPollIntervalSeconds", 300);
+    }
+
+    public String getProxyHost() {
+        return config.getString("eurekaServer.proxyHost", null);
+    }
+
+    public String getProxyPort() {
+        return config.getString("eurekaServer.proxyPort", null);
+    }
+
+    public String getProxyUserName() {
+        return config.getString("eurekaServer.proxyUserName", null);
+    }
+
+    public String getProxyPassword() {
+        return config.getString("eurekaServer.proxyPassword", null);
+    }
+    
+    public boolean shouldGZipContent(){
+        return config.getBoolean("eurekaServer.gzipContent", true);
+    }
+
+    public int getEurekaServerReadTimeoutSeconds(){
+        return config.getInteger("eurekaServer.readTimeout", 8);
+    }
+
+    public int getEurekaServerConnectTimeoutSeconds(){
+        return config.getInteger("eurekaServer.connectTimeout", 5);
+    }
+
+    public String getBackupRegistryImpl() {
+        return config.getString("eurekaServer.backupregistry", null);
```

Comment:
If key names are changed, why not align them with method names and use archaius proxy?